### PR TITLE
Added initial support to support HTML formatting.

### PR DESCRIPTION
**Pull request by welguisz**
**Goal**:  Support HTML Mark-up for sub, sup, b, i, u in individual cells
**Method**: 
*  Read through shared_strings
*  If element is t, it is the same as text
*  If element is r, it is formatted text
** rPr contains the formatting options
** t contains the text
* Set up hash for different elements ... default value for each key is *false*
* if rPr contains the following change hash[key] to *true*

| rPr element | hash key |
|---------|--------|
|b|b|
|i|i|
|u|u|
|vertAlign="subscript"|sub|
|vertAlign="superscript"|sup|
* Write an html string which a user can use Nokogiri::XML to convert from string to XML/HTML

**Known Issue**:
* Possible speed issue.

### CODE REVIEW

File: lib/roo/excelx/shared_strings.rb [hunk 1]
```diff
@@ -11,6 +11,22 @@ def to_a
...
 
+      def to_html
+        @html ||= extract_html
+      end
+
+      # Use to_html or to_a for html returns
+      def use_html?(index)
+        ret_val = false
+        return false if to_a[index].nil?
+        return true if to_html[index].include?('<sub>')
```
Comment (stevendaniels) [inline lib/roo/excelx/shared_strings.rb]: better to use a regexp here, e.g.

    to_html[index][/<([biu]|sub|sup)>/]

**Comment by stevendaniels** [comment]
Is it possible to add more tests for these situations?

File: lib/roo/excelx/shared_strings.rb (hunk 1 above)
Comment (welguisz) [inline lib/roo/excelx/shared_strings.rb]: Will change.

**Comment by welguisz** [comment]
I will  write some unit tests this weekend and add them to git.

**Comment by welguisz** [comment]
Everything looks good from my side.  Thanks,

**Comment by welguisz** [comment]
@stevendaniels Anything that I can do to push  this along.   I  know that  some of my co-workers and developers are looking toward this.

Also, sometime next week I was  thinking of adding formatting to the  output stream so that `Mike<sub>drop</sub>` would display as Mike<sub>drop</sub> in  Excel cell.

**Comment by ilyasgaraev** [comment]
@stevendaniels @welguisz Hi guys! Could you please let me know what state this pull request is in? Are you planning to merge it, and if so, when? Thanks!

**Comment by welguisz** [comment]
@ilyasgaraev ... It is in a state to be merged.  I have not had time to do an output to Excel Cell  yet.  The import of the HTML tags is ready.  Up to  @stevendaniels to decide when he wants to pull  it in.  I don't have the ability to do a merge to the main  repository.

**Comment by stevendaniels** [comment]
I'll be releasing a new version soon.

**Comment by ewmanthei** [comment]
Would it be possible to add an option to disable this feature?  Something like:
`xlsx = Roo::Spreadsheet.open('./myfile.xlsx', injectHtmlWrapper: false)`

**Comment by kaluznyo** [comment]
This features must be with option to disable. I try to import to my project a XLSX with html. My purpose is to get html like the excel, but roo wrap all my cell content with <html>. Modify content is not very good

**Comment by welguisz** [comment]
@kaluznyo Here is the pull request for [Disable HTML Injection](https://github.com/roo-rb/roo/pull/392)

**Pull request merged**